flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -131,16 +131,6 @@ class PFTBuilder {
   }
 
   // Get rid of production wrapper
-  bool Pre(const parser::UnlabeledStatement<parser::ForallAssignmentStmt>
```

Comment:
Hunh?

Reply:
Now that this code is actually being lowered, it turns out that this visit is best done at the full statement level, and this special case sub-statement visitor is dead code.  That can be verified with -fdebug-dump-pre-fir output.  It might be possible to do something similar with the IF statement, but that is less constrained than this, so it might be more difficult.

Reply:
Thanks.